packages/autorest.python/autorest/codegen/serializers/sample_serializer.py

```diff
@@ -0,0 +1,144 @@
+# pylint: disable=too-many-lines
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import logging
+from typing import Dict, Any
+from jinja2 import Environment
+
+from autorest.codegen.models.credential_types import AzureKeyCredentialType
+from autorest.codegen.models.credential_types import TokenCredentialType
+from autorest.codegen.models.imports import FileImport, ImportType
+from autorest.codegen.models.operation import OperationBase
+from autorest.codegen.models.operation_group import OperationGroup
+from autorest.codegen.serializers.import_serializer import FileImportSerializer
+from ..models import CodeModel
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SampleSerializer:
+    def __init__(
+        self,
+        code_model: CodeModel,
+        env: Environment,
+        operation_group: OperationGroup,
+        operation: OperationBase[Any],
+        sample: Dict[str, Any],
+        file_name: str,
+        sample_origin_name: str,
+    ) -> None:
+        self.code_model = code_model
+        self.env = env
+        self.operation_group = operation_group
+        self.operation = operation
+        self.sample = sample
+        self.file_name = file_name
+        self.sample_origin_name = sample_origin_name
+
+    def _imports(self) -> FileImportSerializer:
+        imports = FileImport()
+        namespace = (self.code_model.options["package_name"] or "").replace(
+            "-", "."
+        ) or self.code_model.namespace
+        imports.add_submodule_import(
+            namespace, self.code_model.client.name, ImportType.THIRDPARTY
+        )
+        credential_type = getattr(self.code_model.credential, "type", None)
+        if isinstance(credential_type, TokenCredentialType):
+            imports.add_submodule_import(
+                "azure.identity", "DefaultAzureCredential", ImportType.THIRDPARTY
+            )
+        elif isinstance(credential_type, AzureKeyCredentialType):
+            imports.add_import("os", ImportType.STDLIB)
+            imports.add_submodule_import(
+                "azure.core.credentials", "AzureKeyCredential", ImportType.THIRDPARTY
+            )
+        return FileImportSerializer(imports, True)
+
+    def _client_params(self) -> Dict[str, Any]:
+        # client params
+        special_param = dict()
```

Comment:
I think it would be cleaner if we just move this code to the jinja template. It's more confusing with a dictionary when we can jsut loop through the parameters and give them a value in the template

Reply:
@iscai-msft similar reason: https://github.com/Azure/autorest.python/pull/1505#discussion_r994268169